File: page-objects/js-mocha-chai/complete/pages/hotel-detail-page.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

var BookingDetailsPage = require('./booking-details-page.js');

HotelDetailPage = function HotelDetailPage(driver) {
    this.driver = driver;
    this.bookNowButtons = By.css("button.btn.btn-action.btn-block.chk");
};

HotelDetailPage.prototype.bookFirstRoom = function() {
    this.driver.findElements(this.bookNowButtons).then(function(bookNowButtons) {
        bookNowButtons[0].isDisplayed().then(function(isDisplayed) {
            if (!isDisplayed) {
                var actions = new webDriver.ActionSequence(driver);
                actions.mouseMove(bookNowButtons[0]);
                actions.perform();
            }
        });
        bookNowButtons[0].click();
    });
    return new BookingDetailsPage(this.driver);
};

module.exports = HotelDetailPage;

File: page-objects/java-testng/complete/src/main/java/pages/HotelDetailPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.util.List;

@SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
public class HotelDetailPage {
    private WebDriver webDriver;

    @FindBy(css = "button.btn.btn-action.btn-block.chk")
    private List<WebElement> bookNowButtons;

    public HotelDetailPage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public BookingDetailsPage bookFirstRoom() {
        if (!bookNowButtons.get(0).isDisplayed()) {
            Actions actions = new Actions(webDriver);
            actions.moveToElement(bookNowButtons.get(0));
            actions.perform();
        }
        bookNowButtons.get(0).click();
        return new BookingDetailsPage(this.webDriver);
    }

}
